### PR TITLE
A few tweaks to macro docs

### PR DESCRIPTION
**Pull request by mit-mit**
Fix a few spelling mistakes and clarify steps to try the JSON macro out.

### CODE REVIEW

**Comment by mit-mit** [comment]
@davidmorgan per your feedback

**Review by parlough — APPROVED** [review]

File: src/content/language/macros.md [hunk 1]
```diff
@@ -68,18 +69,19 @@ and a `fromJson` deserialization constructor.
...
-6. Run your project with the experiment flag:
+7. Run your project with the experiment flag:
 
    ```console
    dart run --enable-experiment=macros bin/my_app.dart
    ```
 
+[SDK constraint]: /tools/pub/pubspec#sdk-constraints
 [Add the package]: /guides/packages
 [Add the experiment]: /tools/experiment-flags#using-experiment-flags-with-the-dart-analyzer-command-line-and-ide
```
Comment (MaryaBelanger) [inline src/content/language/macros.md:86]:
```suggestion
[Enable the experiment]: /tools/experiment-flags#using-experiment-flags-with-the-dart-analyzer-command-line-and-ide
```

Comment (MaryaBelanger) [reply src/content/language/macros.md:86]: To go with the change on line 63

**Review by MaryaBelanger — APPROVED** [review]
Approve with changes, thanks parker and mit!

**Pull request merged**